### PR TITLE
method to covert static subset to dynamic subset

### PR DESCRIPTION
**Pull request by rkvinoth**
Accommodating https://github.com/cubewise-code/tm1py/issues/262#issue-648862842

### CODE REVIEW

File: TM1py/Services/SubsetService.py [hunk 1]
```diff
@@ -95,6 +95,28 @@ def update(self, subset: Subset, private: bool = False, **kwargs) -> Response:
...
+        """
+        import json
+        from collections import OrderedDict
+        hierarchy_name = hierarchy_name if hierarchy_name else dimension_name
+        payload = OrderedDict()
+        payload['Name'] = subset_name
+        payload['MakePrivate'] = True if private else False
+        payload['MakeStatic'] = True
+        subsets = "PrivateSubsets" if private else "Subsets"
+        request = "/api/v1/Dimensions('{}')/Hierarchies('{}')/{}('{}')/tm1.SaveAs".format(
```
Comment (MariusWirtz) [inline TM1py/Services/SubsetService.py:116]: Please use the `format_url` method here. It takes care of potential single quotes in object names.

**Comment by MariusWirtz** [comment]
@rkvinoth Thanks for implementing it so quickly!
It looks great. Good idea to use the `SaveAs` action on the subset.

I just got one request. 
It would be nice to have a test case for a public and a private subset in the Tests/Subsets.py module.

File: TM1py/Services/SubsetService.py (hunk 1 above)
Comment (rkvinoth) [reply TM1py/Services/SubsetService.py]: oh yeah, my local version wasn't up to date. Just updated it!

**Comment by rkvinoth** [comment]
I will try to chip in the tests in sometime.

File: Tests/Subset.py [hunk 2]
```diff
@@ -310,6 +310,32 @@ def test_update_subset_dynamic_public(self):
...
+            dimension_name=self.dynamic_subset.dimension_name,
+            private=private)
+        self.assertTrue(subset.is_dynamic)
+        self.tm1.dimensions.hierarchies.subsets.make_static(
+            subset_name=self.dynamic_subset.name,
+            dimension_name=self.dynamic_subset.dimension_name,
+            private=private)
+        self.assertTrue(subset.is_static)
+        
+    def test_make_static_private(self):
```
Comment (MariusWirtz) [inline Tests/Subset.py:327]: I think this should be called `test_make_static_public`

File: Tests/Subset.py [hunk 3]
```diff
@@ -310,6 +310,32 @@ def test_update_subset_dynamic_public(self):
...
+        subset = self.tm1.dimensions.subsets.get(
+            subset_name=self.dynamic_subset.name,
+            dimension_name=self.dynamic_subset.dimension_name,
+            private=private)
+        self.assertTrue(subset.is_dynamic)
+        self.tm1.dimensions.hierarchies.subsets.make_static(
+            subset_name=self.dynamic_subset.name,
+            dimension_name=self.dynamic_subset.dimension_name,
+            private=private)
+        self.assertTrue(subset.is_static)
```
Comment (MariusWirtz) [inline Tests/Subset.py:325]: You must retrieve the subset again before you do the check:

``` python
        subset = self.tm1.dimensions.subsets.get(
            subset_name=self.dynamic_subset.name,
            dimension_name=self.dynamic_subset.dimension_name,
            private=private)
        self.assertTrue(subset.is_static)
```

File: Tests/Subset.py [hunk 4]
```diff
@@ -310,6 +310,32 @@ def test_update_subset_dynamic_public(self):
...
```
Comment (MariusWirtz) [inline Tests/Subset.py:338]: Here you must also retrieve the subset again before the check

File: TM1py/Services/SubsetService.py [hunk 5]
```diff
@@ -95,6 +95,28 @@ def update(self, subset: Subset, private: bool = False, **kwargs) -> Response:
...
+        from collections import OrderedDict
+        hierarchy_name = hierarchy_name if hierarchy_name else dimension_name
+        payload = OrderedDict()
+        payload['Name'] = subset_name
+        payload['MakePrivate'] = True if private else False
+        payload['MakeStatic'] = True
+        subsets = "PrivateSubsets" if private else "Subsets"
+        request = format_url("/api/v1/Dimensions('{}')/Hierarchies('{}')/{}('{}')/tm1.SaveAs", dimension_name,
+                             hierarchy_name, subsets, subset_name)
+        return self._rest.POST(request=request, data=json.dumps(payload))
```
Comment (MariusWirtz) [inline TM1py/Services/SubsetService.py:118]: here it should say `url=request` or `url=url`.

In one of the recent refactoring exercises, we changed the argument name from `request` to `url`, in the POST, GET, PATCH, DELETE methods.

**Comment by MariusWirtz** [comment]
@rkvinoth 
regarding the TM1py tests, if you use PyCharm running the tests is quite easy and it gives you confidence that your changes didn't break anything.

In PyCharm you can just right-click the Tests/subsets.py module and select `Run pytest in Subset.py`
Then you you get the below response. If you want to run all tests in TM1py you can right-click the Tests folder and select `Run TM1py Tests`.

image

File: Tests/Subset.py (hunk 2 above)
Comment (rkvinoth) [reply Tests/Subset.py]: sorry, taken care.

File: Tests/Subset.py (hunk 3 above)
Comment (rkvinoth) [reply Tests/Subset.py:329]: sorry, taken care.

File: Tests/Subset.py (hunk 4 above)
Comment (rkvinoth) [reply Tests/Subset.py:346]: sorry, taken care.

File: TM1py/Services/SubsetService.py (hunk 5 above)
Comment (rkvinoth) [reply TM1py/Services/SubsetService.py]: Updated and tested with the latest GIT repo changes

**Comment by rkvinoth** [comment]
I guess we can merge the PR now
image

**Comment by MariusWirtz** [comment]
Will merge.
Appreciate your contribution!

**Pull request merged**